horreum-backend/src/main/resources/db/changeLog.xml

```diff
@@ -2158,5 +2158,42 @@
             CREATE TRIGGER after_run_update AFTER INSERT OR UPDATE OF data ON run FOR EACH ROW EXECUTE FUNCTION after_run_update_func();
         </sql>
     </changeSet>
+
+    <changeSet id="52" author="jwhiting">
+        <createSequence sequenceName="dataset_id_seq" startValue="1" incrementBy="1" cacheSize="1" />
+        <createTable tableName="dataset">
+            <column name="id" type="integer">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+            <column name="data" type="jsonb">
+                <constraints nullable="false" />
+            </column>
+            <column name="start" type="timestamp without time zone">
+                <constraints nullable="false" />
+            </column>
+            <column name="stop" type="timestamp without time zone">
+                <constraints nullable="false" />
+            </column>
+            <column name="description" type="text" />
+            <column name="testid" type="integer">
+                <constraints nullable="false" />
+            </column>
+            <column name="run_id" type="integer">
+                <constraints nullable="false" />
+            </column>
+            <column name="owner" type="text">
+                <constraints nullable="false" />
+            </column>
+            <column name="access" type="integer">
+                <constraints nullable="false" />
+            </column>
+            <column name="token" type="text" />
+            <column name="trashed" type="boolean" defaultValue="false" />
+        </createTable>
+        <sql>
+            GRANT ALL ON SEQUENCE dataset_id_seq TO "${quarkus.datasource.username}";
+            GRANT SELECT, INSERT, DELETE, UPDATE ON TABLE dataset TO "${quarkus.datasource.username}";
```

Comment:
You should also `ALTER TABLE dataset ENABLE ROW LEVEL SECURITY;` and create policies checking access to the table.

Reply:
Yes.

Reply:
You are missing the policy for reading datasets:
```
CREATE POLICY dataset_select ON dataset FOR SELECT
    USING (can_view2(access, owner) OR has_role('horreum.system'));
```